### ISSUE TITLE
letter spacing issue

### ISSUE THREAD

**Issue by webdeveloper0429** [marcbachmann/node-html-pdf#487]
after generate pdf, letter spacing is incorrect.

http://prntscr.com/llofjg

**Comment by Xtrazyx** [comment]
Absolutely. Same problem. Just using Arial Font. The bug appears to be independent of the css.

**Comment by BenjaminDish** [comment]
I also run into the same issue, using Arial.

**Comment by hanchan07** [comment]
Noticing horrible spacing issues with "T" and "C"  T has negative kerning it seems as it is right up to the letter that comes after it, and C has too much, it seems there is almost a space worth between it and the next letter.

**Comment by mayurvir** [comment]
I'm facing a similar issue, where the pdf gets generated absolutely fine on local machine but has weird spacing issues on ubuntu server. Any updates yet?

**Comment by vplugues** [comment]
OMG! I am facing the SAME issue. My ubuntu server is having these weird spacings with different letter combinations, but MY machine (macOS High Sierra) is working just fine with no spacing issues. I noticed that this started to happen when I switched to font Avenir. I will check if switching to the previous font fixes this spacing issue. and Follow back up.

**Comment by anuj-cfra** [comment]
I am having the same issue. It works fine with default fonts but if I try to use some other fonts, this issue came up.

I am using https://fonts.google.com/specimen/Source+Sans+Pro?selection.family=Open+Sans font and the output spacing is not correct. See the below text

<img width="320" alt="Screen Shot 2019-03-18 at 2 57 58 PM" src="https://user-images.githubusercontent.com/47008786/54520335-7b0acd80-498e-11e9-994a-2efa5ceb2a72.png">

Please let me know if any solution for this

**Comment by maheshatvirtusa** [comment]
Hi @anuj-cfra  , am also facing same issue . did you fount any solution for this font spacing error .

**Comment by vplugues** [comment]
Have you tried just changing the font?

**Comment by maheshatvirtusa** [comment]
With below approach working fine ,  for  custom fonts on AWS Lambda  - htmltopdf .
        I created a fonts directory in my root project fonts directory and placed all of my fonts there. Also in that directory I created a fonts.conf file that looks like this:

**Comment by maheshatvirtusa** [comment]
[fonts.txt](https://github.com/marcbachmann/node-html-pdf/files/3725183/fonts.txt)

**Comment by diegoolavarria** [comment]
I did what it says in that .txt but didn't work. I have to mention that my app in production is in EB, not in Lambda. Even so, I did this:

1. downloaded a .woff font file in a folder

![image](https://user-images.githubusercontent.com/15364801/71841242-dc836080-30b6-11ea-8104-3eca3ad1f92d.png)

2. In styles.js, I'm bringing the files so the generated HTML receive the corresponding style

```
export default styles => {
  const extraStyles = styles ? styles.map(s => files[s]).join('') : ''

  return `<style>
  @font-face {
    src: url(/fonts/NunitoSans-Regular.woff) format('woff');
    font-family: 'Nunito Sans', sans-serif;
  }
  @font-face {
    src: url(/fonts/NunitoSans-Black.woff) format('woff');
    font-family: 'Nunito Sans', sans-serif;
    font-weight: 900;
  }
  body {
    color: #00171f;
    font-family: 'Nunito Sans', sans-serif;
  }
[...]
```

3. Deploy into EB

The generated PDF still has letter separation issues.

**Comment by GuilhermeHausz** [comment]
Anyone find a solution for this issue?
